assertj-core/src/main/java/org/assertj/core/api/Assertions.java

```diff
@@ -1210,6 +1212,17 @@ public static <T extends Throwable> AbstractThrowableAssert<?, T> assertThat(T a
     return assertThat(catchThrowable(shouldRaiseThrowable)).hasBeenThrown();
   }
 
+  @CanIgnoreReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assertThatThrownBy(ThrowingCallableWithValue shouldRaiseThrowable) {
+    Object value;
+    try {
+      value = shouldRaiseThrowable.call();
+    } catch (Throwable throwable) {
+      return assertThat(throwable);
+    }
+    throw Failures.instance().failure(String.format("Expecting code to raise a throwable, but it returned [%s] instead", value));
+  }
```

Comment:
I am pretty sure that this is not how it should eventually look like.
Very minimum would be a javadoc.
This is for issue illustration / POC at this stage.

Happy to improve if this is at least directionally correct.

Reply:
This looks good to me.

We need to do the same in BDDAssertions